### PR TITLE
feat: send to s3, create local dump, map languague, text_general fields, 

### LINKED ISSUE #277: Send data to S3

### LINKED ISSUE #302: Clear filters from "Undetermined", "Null", empty values
Acceptance criteria:

- [ ]  As a user I want to see "Not specified" field instead off "Undetermined", "UNKNOWN" and "Null"

### LINKED ISSUE #382: Add an option to create local dump while processing data

### LINKED ISSUE #383: Introduce new oag dump

### LINKED ISSUE #389: Add text_general fields
Add:
- author_names_tg
- keywords_tg
- tag_list_tg

### PR DESCRIPTION
**Pull request by Michal-Kolomanski**
Unfortunately, many tickets were solved in this PR. Sorry for that. 

Scope:
- Add possibility to create local dump -> closes #382 
- Add possibility to send dump to s3 -> closes #277 
- Introduce new oag dump -> closes #383 
- Map language -> closes #302 
  Language mapping:
  - `("undetermined", "unknown", "null") `- are mapped (no matter capitalization) to `Not specified`
   - `en` - is mapped to `English`. (trainings)
   - `es` - is mapped to `Spanish`. (trainings)
- Add text_general fields -> closes #389

### CODE REVIEW

File: transform/all_collection/spark/transformations/commons.py [hunk 1]
```diff
@@ -189,7 +216,43 @@ def map_publisher(df: DataFrame) -> DataFrame:
...
+
+        elif isinstance(language, str):
+            for desired_lan, old_lang in language_mapping.items():
+                if language.lower() in old_lang:
+                    language_column.append([desired_lan])
+                    break
+            else:
+                language_column.append([language])
+        else:
+            raise TypeError(f"{language} is not list or str")
```
Comment (danielkryska) [inline transform/all_collection/spark/transformations/commons.py:252]: To deep nesting of `for` loops and `if`'s.

Comment (danielkryska) [reply transform/all_collection/spark/transformations/commons.py:252]: You may rewrite it or leave it. :P

File: transform/all_collection/spark/utils/utils.py [hunk 2]
```diff
@@ -108,14 +109,15 @@ def print_results(failed_files: Dict, logger: Log4J) -> None:
...
-            logger.error(f"{col_name}: {f_file}")
+    for col_name, dests in failed_files.items():
+        for dest, f_file in dests.items():
+            if f_file:
+                if not err_printed:
+                    err_printed = True
+                    logger.error(
+                        f"Certain files failed either to be transformed or to be sent to {dest}"
+                    )
+                logger.error(f"{col_name}: {f_file}, destination: {dest}")
```
Comment (danielkryska) [inline transform/all_collection/spark/utils/utils.py:120]: Very deep nesting, you may fix or leave it. :P

**Review by danielkryska — APPROVED** [review]

File: transform/all_collection/spark/transformations/commons.py (hunk 1 above)
Comment (Michal-Kolomanski) [reply transform/all_collection/spark/transformations/commons.py]: I rewrote that

**Pull request merged**